pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -167,32 +119,110 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 	}
 
 	controlPlaneMachines := GetControlPlaneMachines(clusterMachines)
+	if len(controlPlaneMachines) == 0 {
+		return errors.New("unable to proceed with machine creation until there is at least 1 control plane machine")
+	}
 
-	isNodeJoin, err := a.isNodeJoin(scope, controlPlaneMachines)
-	if err != nil {
-		return errors.Wrapf(err, "failed to determine whether machine %q should join cluster %q", machine.Name, cluster.Name)
+	/*
```

Comment:
Can we add unit tests for this? It will protect us from future violations of today's assumptions 😊

Reply:
Maybe refactor this code into a method that we can write tests against :)

Reply:
(copying my email comments that didn't slot in the right place)

Of course! I only had a limited amount of time today and I wanted this up for review first.